### PR TITLE
CEDS-2758 split carrier details page into two pages

### PR DESCRIPTION
**Pull request by glidester**

### CODE REVIEW

File: test/views/declaration/CarrierEoriNumberViewSpec.scala
Comment (MaciejRewera) [inline test/views/declaration/CarrierEoriNumberViewSpec.scala:116]: Nit: This is something I am always unsure about - whether view specs should test for different possible errors, or maybe it is enough to just check an error. If there is error summary, error text near the field etc., and leave the responsibility to test for all errors to form specs.
This is not a problem, but it's just I've always had this inner conflict.

Comment (glidester) [reply test/views/declaration/CarrierEoriNumberViewSpec.scala:116]: Well not knowing the code that well I just copied what the ConsignorEoriNumberViewSpec did. If this is something we should change then probably should do it for all view specs.

File: test/views/declaration/CarrierEoriNumberViewSpec.scala
Comment (MaciejRewera) [inline test/views/declaration/CarrierEoriNumberViewSpec.scala:46]: Should there not be `onEveryDeclarationJourney` instead? Right now, you are testing it against a single journey only.

File: app/controllers/navigation/Navigator.scala
Comment (MaciejRewera) [inline app/controllers/navigation/Navigator.scala:423]: 👍

**Comment by PauloPortugal** [comment]
You need also to update the https://github.com/hmrc/customs-declare-exports-frontend/blob/master/docs/Customs%20Declare%20Exports%20AutoComplete.js and so do I :)

File: app/forms/declaration/ConsigneeDetails.scala
Comment (MaciejRewera) [inline app/forms/declaration/ConsigneeDetails.scala:29]: 👍

File: test/views/declaration/CarrierEoriNumberViewSpec.scala
Comment (glidester) [reply test/views/declaration/CarrierEoriNumberViewSpec.scala]: Fixed. Added all journey types apart from SUPPLEMENTAL (that does not have Carrier in its journey flow)

File: test/unit/controllers/declaration/CarrierDetailsControllerSpec.scala
Comment (MaciejRewera) [inline test/unit/controllers/declaration/CarrierDetailsControllerSpec.scala:107]: Could this possibly be 'merged' with the clause above? It seems to cover the same set of journeys.

File: test/unit/controllers/declaration/CarrierDetailsControllerSpec.scala
Comment (MaciejRewera) [inline test/unit/controllers/declaration/CarrierDetailsControllerSpec.scala:138]: Should there be `onJourney` for all other journeys?

File: test/views/declaration/CarrierEoriNumberViewSpec.scala
Comment (MaciejRewera) [reply test/views/declaration/CarrierEoriNumberViewSpec.scala:117]: Agree with you. But sadly, taking this approach we are going to probably never actually do it ;-P

File: test/views/declaration/CarrierDetailsViewSpec.scala
Comment (MaciejRewera) [inline test/views/declaration/CarrierDetailsViewSpec.scala:43]: Nit: Just figured out that it could be beneficial to extract the test addresses so they can be re-used accross various *Details specs. No need to do it now, unless you want to.

File: test/views/declaration/CarrierDetailsViewSpec.scala
Comment (MaciejRewera) [inline test/views/declaration/CarrierDetailsViewSpec.scala:118]: Could this be extracted to not repeat it every test case? Unless you need a new instance every test case.

**Review by MaciejRewera — COMMENTED** [review]
Left a few comments, but other than that it LGTM.
Also, I've noticed there is no update to 'Quick Submit' script for the new pages. Could you update it as well?
I'll have a second review and approve then.

File: test/unit/controllers/declaration/CarrierDetailsControllerSpec.scala
Comment (glidester) [reply test/unit/controllers/declaration/CarrierDetailsControllerSpec.scala]: Done

File: test/unit/controllers/declaration/CarrierDetailsControllerSpec.scala
Comment (glidester) [reply test/unit/controllers/declaration/CarrierDetailsControllerSpec.scala]: Added for all other journey types

File: test/views/declaration/CarrierDetailsViewSpec.scala
Comment (glidester) [reply test/views/declaration/CarrierDetailsViewSpec.scala]: Yes, good spot! Amended

File: test/views/declaration/CarrierDetailsViewSpec.scala
Comment (glidester) [reply test/views/declaration/CarrierDetailsViewSpec.scala:43]: If I was more comfortable with the code base and hadn't already spent a couple of days on this then I would. One for the future me thinks ;)

File: docs/Customs Declare Exports AutoComplete.js
Comment (MaciejRewera) [inline docs/Customs Declare Exports AutoComplete.js:367]: Doesn't the url end with `carrier-address` ?

**Review by MaciejRewera — DISMISSED** [review]
LGTM

**Review by MaciejRewera — APPROVED** [review]
Great stuff. LGTM.

**Pull request merged**
